### ISSUE TITLE
Pod for OTel Collector doesn't start when deployed to K8S using --values my-values-file.yaml

### ISSUE THREAD

**Issue by kainawroth** [open-telemetry/opentelemetry-demo#964]
# Bug Report

OpenTelemetry Demo v1.4.0 -- OTel Collector

## Symptom

When I deploy the demo to a K8S cluster (AKS) using custom values via `helm install my-otel-demo open-telemetry/opentelemetry-demo --values my-values-file.yaml`, the pod for the OTel Collector doesn't start.

This is also the case if I copy the exact values from _values.yaml_ found in _opentelemetry-demo-0.22.4.tgz_ or https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-demo/values.yaml.

## Investigate

After running `kubectl logs my-otel-demo-new-otelcol-565b7f8575-cn9hg`, I get the following:

_Error: invalid configuration: service::pipeline::metrics: references receiver "spanmetrics" which is not configured 
2023/07/07 23:22:52 collector server run finished with error: invalid configuration: service::pipeline::metrics: references receiver "spanmetrics" which is not configured_

and 

_Error: invalid configuration: service::pipeline::traces: references exporter "spanmetrics" which is not configured 
2023/07/07 23:30:03 collector server run finished with error: invalid configuration: service::pipeline::traces: references exporter "spanmetrics" which is not configured_

## Solution

When I remove `spanmetrics` from _service > pipelines > traces > exporters_ and from _service > pipelines > metrics > receivers_, the pod starts without any issues.

## Reproduce

1. Create _my-values-file.yaml_ file
2. Copy values from https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-demo/values.yaml
3. Install demo via `helm install my-otel-demo open-telemetry/opentelemetry-demo --values my-values-file.yaml`

## Additional Context

If I install the demo without `--values`, this problem doesn't occur and the pod for the OTel Collector starts.

I know spanmetrics was mentioned as a breaking change in the release notes for 1.4.0 in https://github.com/open-telemetry/opentelemetry-demo/releases, so if this is not a bug but a configuration problem on my end, please change the label or close the bug.

**Side note:** I also had to increase the memory for the Frontend microservice for it to run on kubernetes version 1.25.6 or later, but I don't know if this is an AKS-exlusive issue or happens with all K8S services.

**Comment by mcy567** [comment]
我参考 [Kubernetes deployment ](https://opentelemetry.io/docs/demo/kubernetes-deployment/#expose-services-using-service-type-configurations)，遇到了同样的问题：
```
[root@rzy4813 ~]# kubectl get pods -A
NAMESPACE         NAME                                                  READY   STATUS    RESTARTS        AGE
calico-system     calico-kube-controllers-6b44b54755-ncscl              1/1     Running   1 (12h ago)     14h
calico-system     calico-node-t7q6b                                     1/1     Running   1 (12h ago)     14h
calico-system     calico-typha-7d7f786d85-cw4rs                         1/1     Running   1 (12h ago)     14h
kube-system       coredns-6d4b75cb6d-f8qd7                              1/1     Running   1 (12h ago)     14h
kube-system       coredns-6d4b75cb6d-prs6v                              1/1     Running   1 (12h ago)     14h
kube-system       etcd-rzy4813                                          1/1     Running   9 (12h ago)     14h
kube-system       kube-apiserver-rzy4813                                1/1     Running   8 (12h ago)     14h
kube-system       kube-controller-manager-rzy4813                       1/1     Running   8 (12h ago)     14h
kube-system       kube-proxy-5ngk9                                      1/1     Running   1 (12h ago)     14h
kube-system       kube-scheduler-rzy4813                                1/1     Running   8 (12h ago)     14h
my-otel-demo      my-otel-demo-accountingservice-5486848bdc-kfdq6       1/1     Running   0               13h
my-otel-demo      my-otel-demo-adservice-fd6847f49-gvvp2                1/1     Running   1 (12h ago)     13h
my-otel-demo      my-otel-demo-cartservice-5cd55456f7-fjnct             1/1     Running   0               13h
my-otel-demo      my-otel-demo-checkoutservice-7bc648c948-q5vm2         1/1     Running   0               13h
my-otel-demo      my-otel-demo-currencyservice-7ff57f89b6-d29lf         1/1     Running   0               13h
my-otel-demo      my-otel-demo-emailservice-685d67d98d-mwf7c            1/1     Running   1 (6h18m ago)   13h
my-otel-demo      my-otel-demo-featureflagservice-67c4457d6d-cnd76      1/1     Running   0               13h
my-otel-demo      my-otel-demo-ffspostgres-694bb9c69b-s45cw             1/1     Running   0               13h
my-otel-demo      my-otel-demo-frauddetectionservice-78668c7f59-4wgfs   1/1     Running   0               13h
my-otel-demo      my-otel-demo-frontend-658b979c9d-h5h7b                1/1     Running   0               13h
my-otel-demo      my-otel-demo-frontendproxy-b7c4b9fdf-j6g84            1/1     Running   0               13h
my-otel-demo      my-otel-demo-grafana-86785d6584-bxz6z                 1/1     Running   1 (12h ago)     13h
my-otel-demo      my-otel-demo-jaeger-586fddfbbb-c5cbk                  1/1     Running   1 (12h ago)     13h
my-otel-demo      my-otel-demo-kafka-7d6994895b-r2h2s                   1/1     Running   0               13h
my-otel-demo      my-otel-demo-loadgenerator-9f4cc9d6c-mcwvd            1/1     Running   0               13h
my-otel-demo      my-otel-demo-otelcol-74746d647b-htwfm                 0/1     Error     2 (24s ago)     25s
my-otel-demo      my-otel-demo-paymentservice-58cb9744d9-v7qtb          1/1     Running   0               45s
my-otel-demo      my-otel-demo-productcatalogservice-5b8cdf69b8-pl5tp   1/1     Running   2 (5h7m ago)    13h
my-otel-demo      my-otel-demo-prometheus-server-66cf7fb98b-tm9v7       1/1     Running   0               13h
my-otel-demo      my-otel-demo-quoteservice-575655dd6f-mk8sw            1/1     Running   0               13h
my-otel-demo      my-otel-demo-recommendationservice-55b96499cc-9wthm   1/1     Running   0               13h
my-otel-demo      my-otel-demo-redis-6789679fcf-kn86p                   1/1     Running   1 (12h ago)     13h
my-otel-demo      my-otel-demo-shippingservice-795d76d564-65fnq         1/1     Running   0               13h
tigera-operator   tigera-operator-d7957f5cc-nzcn4                       1/1     Running   1 (12h ago)     14h
[root@rzy4813 ~]# kubectl logs my-otel-demo-otelcol-74746d647b-htwfm --all-containers=true -n my-otel-demo 
Error: invalid configuration: service::pipelines::traces: references exporter "spanmetrics" which is not configured
2023/09/26 02:06:51 collector server run finished with error: invalid configuration: service::pipelines::traces: references exporter "spanmetrics" which is not configured

```

**Comment by puckpuck** [comment]
This should be closed with [#3372](https://github.com/open-telemetry/opentelemetry.io/pull/3372)
